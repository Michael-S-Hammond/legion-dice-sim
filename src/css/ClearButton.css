.clear-button {
  display: flex;
  width: 25px;
  height: 25px;
  background-image: url("../img/x.svg");
  background-repeat: no-repeat;
  background-size: 75%;
  background-position-x: center;
  background-position-y: center; }
