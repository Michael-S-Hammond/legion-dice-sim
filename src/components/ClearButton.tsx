import '../css/ClearButton.css';

import React from 'react';

type ClearButtonProps = {
    onClick: () => void,
    tooltip: string,
};

class ClearButton extends React.Component<ClearButtonProps> {
    constructor(props : ClearButtonProps) {
        super(props);
    }

    render() : JSX.Element {
        return (
            <button
                className="btn btn-secondary border rounded-lg clear-button mx-2 my-auto"
                onClick={() => this.props.onClick()}
                title={this.props.tooltip}>
            </button>
        );
    }
}

export default ClearButton;
