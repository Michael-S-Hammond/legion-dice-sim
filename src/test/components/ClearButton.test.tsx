import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import ClearButton from '../../components/ClearButton';

describe('ClearButton', () => {
    it('matches the snapshot', () => {
        function onClick() {
        }

        const counter = <ClearButton
                onClick={onClick}
                tooltip='Clear...'
            ></ClearButton>
        const snapshot = create(counter);
        expect(snapshot.toJSON()).toMatchSnapshot();
    });

    it('handles being clicked', () => {
        const onButtonClickMock = jest.fn();

        let wrapper = shallow(<ClearButton
            onClick={onButtonClickMock}
            tooltip='Clear...'
        ></ClearButton>);
        wrapper.find('.btn').simulate('click');

        expect(onButtonClickMock).toHaveBeenCalledTimes(1);
    })
});
