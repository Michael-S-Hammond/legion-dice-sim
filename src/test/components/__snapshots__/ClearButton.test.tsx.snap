// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton matches the snapshot 1`] = `
<button
  className="btn btn-secondary border rounded-lg clear-button mx-2 my-auto"
  onClick={[Function]}
  title="Clear..."
/>
`;
